### DISCUSSION TITLE
Confusion about duplication after Keyframing/Extending

### DISCUSSION THREAD

**Discussion by AliKarpuzoglu** [keiwando/toonsquid#652] [Q&A] [answered]
So I have started with animation using Toonsquid yesterday and really enjoyed it. 
I have tried doing an animation with keyframes by extending a frame, and then moving the objects. The animation looks good enough for me, however: Now I want to continue the animation without the keyframes. I have thought about just duplicating a frame and continuing from there. However, whenever I duplicate the frame, I get a duplicate of the first frame, before anything has moved.

Imagine Frame 1: A ball all the way on the left.
Frame 10: a ball all the way on the right, keyframed to move there. 

The ball moves from left to right. On duplication at Frame 10, I get a drawing of the ball all the way on the left. And if I copy all the layers at the last frame and paste it into a new drawing, I still get the first frames location.

Thanks. I have been looking around in both the guides and multiple youtube tutorials, as well as the forum here. However, maybe there‘s just a misunderstanding on my part as to how this is supposed to work

**Comment by Cat-Butler** [comment, marked as answer]
ToonSquid is not purely a frame-by-frame animation app like some simple animation apps. Once you start using keyframes, each frame is no longer a standalone image.

There are a couple of ways you can approach thinking about this.

One way is to embrace the digital keyframing approach:
There are no frames, only drawings and keyframes. Frames are a measure of time, similar to seconds (but smaller). So, a drawing can show on camera for 10 frames (instead of 10 seconds), or 1 frame (instead of one second).
Keyframes are instructions to position the drawing.
So you can have a drawing of a ball that shows for 10 frames, and you use keyframes to describe where the drawing is positioned at each moment in time - but it is all the same drawing.
Then you have a new drawing at (the time of) frame 11 that only shows for 1 frame.

The position of the ball is not a property of the drawing (the ball), it is a property of the keyframe.
So, if you copied the same drawing, your ball at frame 11 would also require copying the keyframe from frame 10 (copying and pasting the little diamond in the timeline, in the Position channel, from frame 10 to frame 11).
Otherwise, to avoid using keyframes in frame 11, you have two choices:
Either completely re-draw the ball at frame 11, probably using the onion skin feature to help align it.
Or, copy then edit the ball drawing (as originally showing in frame 1, unless you moved that also), probably using the onion skin feature to help align it.

Another way is to think of ToonSquid is in the manner of the old cel style of editing:
You have moved the ball drawing (cel) and at each frame (similar to each second, but smaller) the film camera takes a frame "photograph".
To get the ball to show at the same position (as frame 10) in frame 11, you have the same 3 choices:
- Put the cel copy in the same position relative to the "camera" (copy the frame 10 keyframe to frame 11).
- Copy and edit the drawing (edit the copy of the cel/drawing from frame 1 to move the ball as an edit rather than as a keyframe, similar to erasing and re-drawing the ball on the physical cel)
- Re-draw the ball completely (create a new cel)

![KeyframeCel](https://github.com/user-attachments/assets/ba114371-d8ab-41e4-a3de-cd68954b04a3)

Note: you can also do what you originally wanted, convert those first 10 frames into single-frame drawings, by Rasterizing the timeline layer for the ball layer (tap the layer name at the end of the timeline and chose Rasterize from the pop-up menu).
You can then easily copy/duplicate frame 10 to frame 11 and the ball will stay in the same place.
_However:_
If you do this you will _lose any drawing layers_ and any _vectors will be converted to pixels_.
It's similar to printing out all the film frames from the film camera onto new individual cels.
Each frame in the timeline layer will become a static 1-frame pixel drawing, as if you repainted each one.

**Reply by AliKarpuzoglu** [reply]
wow, thank you for the detailed explanation!
